bottlecap/src/otlp/agent.rs

```diff
@@ -127,9 +134,20 @@ impl Agent {
     }
 
     async fn v1_traces(
-        State((config, tags_provider, processor, trace_processor, trace_tx)): State<AgentState>,
+        State((config, tags_provider, processor, trace_processor, trace_tx, api_key_factory)): State<AgentState>,
         request: Request,
     ) -> Response {
+        let Some(api_key) = api_key_factory.get_api_key().await else {
```

Comment:
Abort at the beginning of `v1_traces` API handler

Reply:
Is it possible that the customer's code calls `/v1/traces` api synchronously, and we slow down the customer's Lambda by doing the heavy operation of resolving api key here?
If so, it might be better to further defer key resolution by moving it out of the API handler.